File: AngularTask/src/app/Project/student-subject/student-subject.ts
import { Student } from 'src/app/Project/student';
import { Subject } from 'src/app/Project/subject';

export class StudentSubject {
    constructor(
        public StudentId:number, 
        public SubjectId:number,
        public Grade?:string, 
        public student?: Student,
        public subject? : Subject
        ){}
}

File: Angular_API/Angular_API/Models/StudentSubject.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace Angular_API.Models
{
    public class StudentSubject
    {
        [Column(Order =1)]
        [Key]
        [ForeignKey("student")]
        public int StudentId { get; set; }

        [Column(Order = 2)]
        [Key]
        [ForeignKey("subject")]
        public int SubjectId { get; set; }

        public string Grade { get; set; }

        public subject subject { get; set; }
        public Student student { get; set; }

    }
}